### PR TITLE
Sam/mergetrails

### PR DESCRIPTION
**Pull request by samlawhon**
* Added merging trails functionality
* Added drawing trails functionality

### CODE REVIEW

File: api/flaskr/elevation.py [hunk 1]
```diff
@@ -18,8 +18,14 @@ def get_elevation(coords):
     """
     locations_string = ""
 
-    for coord in coords:
-        locations_string += f"{coord['lat']},{coord['lng']}|"
+    # try only analyzing every other coordinate if the request is over 500 coords (Google maps elevation API limit)
+    if len(coords) > 500:
```
Comment (tdipadova3rd) [inline api/flaskr/elevation.py:22]: Make 500 a var, no magic numbers

File: api/flaskr/elevation.py [hunk 2]
```diff
@@ -29,14 +35,17 @@ def get_elevation(coords):
...
     endpoint = "https://maps.googleapis.com/maps/api/elevation/json"
-
-    result = requests.get(endpoint, payload).json()
-
-    return result["results"]
+    
+    # if there are still too many coordinates, raise an error (route map quality may start to degrade once we reduce coords by a third)
+    try:
+        result = requests.get(endpoint, payload).json()
+        return result["results"]
```
Comment (tdipadova3rd) [inline api/flaskr/elevation.py:42]: You should probably also check return code to make sure it was successful (200s) and not rejected (400s) or error (500s)

File: api/flaskr/join_routes.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+from api.flaskr.great_circle import great_circle
+from sys import maxsize
+from queue import PriorityQueue
+
+def join_routes(routes):
```
Comment (tdipadova3rd) [inline api/flaskr/join_routes.py:5]: This function is too long, break into sub-functions

File: api/flaskr/elevation.py (hunk 1 above)
Comment (samlawhon) [reply api/flaskr/elevation.py:22]: NO MAGIC NUMBERS!!

**Review by tdipadova3rd — APPROVED** [review]

**Pull request merged**